vars/customWARPackager.groovy

```diff
@@ -0,0 +1,117 @@
+/**
+ * Builds a custom Jenkins WAR using <a href="https://github.com/jenkinsci/custom-war-packager">Custom WAR Packager</a>.
+ * The method supports both YAML BOM and Custom WAR Packager YAM as an input.
+ *
+ * @param metadataFile Path to the metadata file.
+ * @param outputWAR Path, to which the output WAR file should be written
+ * @param outputBOM Path, to which the output BOM file should be written.
+ *                  More info about BOMs: https://github.com/jenkinsci/jep/pull/92
+ * @param mvnSettingsFile
+ *                  Maven Settings file to be used during the build.
+ *                  If {@code null}, it will be determined automatically.
+ * @return nothing
+ */
+def build(String metadataFile, String outputWAR, String outputBOM, String mvnSettingsFile) {
+    def metadata = readYaml(file: metadataFile)
+    def bomFilePath = metadata.packaging.bom ?: null
+    def environment = metadata.packaging.environment
```

Comment:
This is open to a NPE in case no `packaging` section is found, I would check for that section first and immediately return here if not found or use a NPE safe operator

Reply:
Agreed